src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/JsonFormat/JsonFormatFactorySearchResultPackage.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class JsonFormatFactorySearchResultPackage
+    {
+        public static ISearchResultPackage GetPackage(IPackageSearchMetadata packageSearchMetadata, PackageSearchVerbosity verbosity, bool exactMatch, string deprecation)
+        {
+            if (exactMatch)
```

Comment:
What's the difference between exact match and non-exact match in terms of rendering?
Basically why do we need the indirection that leads to 3 extra types for exact match?

Reply:
Exact-match operation in a sense is totally different from non-exact match operation. Basically, exact match lists all the available versions for a package. However, a non-exact-match operation on the other hand looks up a search phrase and lists available packages. Rendering the otput from the two can be a simillar operation because in both cases we list packages. Howeve, the only difference comes in when we are rendering the version property. For exact match, the property name is "version", however, for non-exact match operations, it is "latestVersion".